src/components/Utilities.tsx

```diff
@@ -342,17 +334,80 @@ export function MapFilterSmallComponent() {
                     />
                   </form>
                 </div>
-            </section>
-
-            <div className="fixed w-full px-6 bg-white border-t border-slate-300 bottom-0 py-3">
-              <Button className="bg-brand-primary hover:bg-blue-600 w-full"
-              onClick={handleFilter}>
-                Apply All
-              </Button>
+              </section>
+
+              <div className="fixed w-full px-6 bg-white border-t border-slate-300 bottom-0 py-3">
+                <Button
+                  className="bg-brand-primary hover:bg-blue-600 w-full"
+                  onClick={handleFilter}
+                >
+                  Apply All
+                </Button>
+              </div>
             </div>
           </div>
+        </motion.div>
+      </section>
+    </>
+  );
+}
+
+type TMapProps = {
+  data: Post[];
+  className?: string;
+};
+
+const DynamicMapSM = dynamic(() => import("@/components/MapComponent"), {
+  ssr: false,
+});
+export function MapFilterSmallComponent({ data, className }: TMapProps) {
+  const [toggleFilter, setToggleFilter] = useState<boolean>(false);
+  const [toggleMap, setToggleMap] = useState(false);
+
+  function handleToggleFilter() {
+    setToggleFilter((curState) => !curState);
+  }
+
+  function handleToggleMap() {
```

Comment:
You need to remove the map when the map is closed to avoid map initialization error. Hint: You can check to see if the map is truthy or falsy to then you can call the remove method on the map instance.

Reply:
Alright Sir, and thanks for the hint. Will make the changes ASAP